DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -28,9 +28,16 @@ internal class DataUploadWorker: DataUploadWorkerType {
     private let contextProvider: DatadogContextProvider
     /// Delay used to schedule consecutive uploads.
     private let delay: DataUploadDelay
-
-    /// Upload work scheduled by this worker.
+    /// Maximum number of batches to upload in one request.
+    private let maxBatchesPerUpload: Int
+
+    /// Batch reading work scheduled by this worker.
+    @ReadWriteLock
+    private var readWork: DispatchWorkItem?
+    /// Batch upload work scheduled by this worker.
+    @ReadWriteLock
     private var uploadWork: DispatchWorkItem?
```

Comment:
/question I do prefer this than a boolean for cancellation 👍 I wonder why not using a single work-item for both? we would then have just one to cancel.

Reply:
When I started the clean up it felt a little more consistent to me.

In the end of the day it's closure vs DispatchWorkItem

WDYT?

Reply:
Yes, not a big deal!